ddtrace/internal/runtime/runtime_metrics.py

```diff
@@ -86,6 +88,7 @@ def __attrs_post_init__(self):
         self._dogstatsd_client = get_dogstatsd_client(self.dogstatsd_url or ddtrace.internal.agent.get_stats_url())
         self.tracer = self.tracer or ddtrace.tracer
         self.tracer.on_start_span(self._set_language_on_span)
+        self.interval = self.interval or _get_default_interval()
```

Comment:
I'm a bit confused about why you need this, since you have `factory` being set with `attr.ib`?

Reply:
I think the real problem is in `RuntimeWorker.enable`:
~~~ python
            runtime_worker = cls(flush_interval, tracer, dogstatsd_url)  # type: ignore[arg-type]
~~~
This is passing `None` to `_interval` by default. Perhaps this should be fixed by defining a `*DEFAULT_INTERVAL*` constant and do

~~~ python
    def enable(cls, flush_interval=*DEFAULT_INTERVAL*, tracer=None, dogstatsd_url=None):
~~~

Reply:
Oh right. Yeah using `DEFAULT_INTERVAL` would be a good pick.

Reply:
Thanks for pointing this out @jd. The combination of a factory for a default value with a post initialization is unnecessary and misleading. The other attributes default to `None` so a post initialization to set them makes sense.